### ISSUE TITLE
Fish Shell Compatibility

### ISSUE THREAD

**Issue by madsem** [shipping-docker/vessel#123]
Would this be something you would be willing to integrate into Vessel?

**Comment by madsem** [comment]
PS: I would love to make a PR for that, is what I meant, not that you have to do it :)

**Comment by fideloper** [comment]
Hi!

Can you give me a little more information on what would be involved /
what's not currently compatible?

Thanks!

**Comment by madsem** [comment]
Sure!

By looking at the code so far, I think the `vessel` bash script would need to be rewritten as fish script, and then used as an alternative to bash. Fish shell is not bash compatible: https://fishshell.com/docs/current/index.html

Other than that I don't think there would be adjustments necessary, but I would have to run tests of course.

**Comment by madsem** [comment]
Alright, it seems this was just me not knowing some basics :P 

I just learned something I think: You made the `vessel` bash script executable, and because of that the terminal just runs it as bash, no matter what shell you're using. And the ENV vars are set for all shells, i.e I switch from fish to bash and have all the same ENV vars set and vice versa. I did not know that!

I'm gonna close that as it was clearly user error.
